lib/pricing-poller.js

```diff
@@ -136,44 +136,49 @@ class PricingPoller {
 
     for (let region of this.regions) {
       let ec2 = this.ec2[region];
-      let [azResult, priceResult] = await Promise.all([
-        this.runaws(ec2, 'describeAvailabilityZones', {
-          Filters: [{
-            Name: 'state',
-            Values: ['available'],
-          }],
-        }),
-        this.runaws(ec2, 'describeSpotPriceHistory', {
+      let nextToken = undefined;
+      let azResult = await this.runaws(ec2, 'describeAvailabilityZones', {
+        Filters: [{
+          Name: 'state',
+          Values: ['available'],
+        }],
+      });
+      let zones = azResult.AvailabilityZones.map(x => x.ZoneName);
+
+      do {
+        let priceResult = await this.runaws(ec2, 'describeSpotPriceHistory', {
           StartTime: startTime,
+          NextToken: nextToken,
           Filters: [{
             Name: 'product-description',
             Values: ['Linux/UNIX'],
           }],
-        }),
-      ]);
+        });
 
-      // Get the information from the API responses that we care about
-      let zones = azResult.AvailabilityZones.map(x => x.ZoneName);
-      let pricePoints = priceResult.SpotPriceHistory.filter(x => zones.includes(x.AvailabilityZone));
+        nextToken = priceResult.NextToken;
 
-      // Determine what the prices which the API described.
-      let spotPrices = this._findSpotPricesForRegion({pricePoints});
-      let onDemandPrices = this._findOnDemandPricesForRegion();
+        // Get the information from the API responses that we care about
+        let pricePoints = priceResult.SpotPriceHistory.filter(x => zones.includes(x.AvailabilityZone));
 
-      // Add some information to the spot prices
-      spotPrices = spotPrices.map(x => {
-        Object.assign(x, {type: 'spot', region});
-        return x;
-      });
+        // Determine what the prices which the API described.
+        let spotPrices = this._findSpotPricesForRegion({pricePoints});
+        let onDemandPrices = this._findOnDemandPricesForRegion();
 
-      // Add some information to the on-demand prices
-      onDemandPrices = onDemandPrices.map(x => {
-        Object.assign(x, {type: 'ondemand', region});
-        return x;
-      });
+        // Add some information to the spot prices
+        spotPrices = spotPrices.map(x => {
+          Object.assign(x, {type: 'spot', region});
+          return x;
+        });
+
+        // Add some information to the on-demand prices
+        onDemandPrices = onDemandPrices.map(x => {
+          Object.assign(x, {type: 'ondemand', region});
+          return x;
+        });
 
-      Array.prototype.push.apply(prices, spotPrices);
-      Array.prototype.push.apply(prices, onDemandPrices);
+        Array.prototype.push.apply(prices, spotPrices);
```

Comment:
Hmm, I think that since we're really looking for the max price of a given instanceType/region/zone triplet, we probably don't need to build one gigantic list in memory of all requests.  Instead, how about we find the maximum price of each api request, then compare it to the maximum price of the previous api request.

Reply:
Doesn't the call to `_findSpotPricesForRegion` do that?